app/src/main/java/com/github/drawyourpath/bootcamp/userProfileCreation/PersonalInfoFragment.kt

```diff
@@ -0,0 +1,218 @@
+
+        //validate and store data if they are correct, otherwise print some error message to the user
+        val validatePersonalInfoButton: Button =
+            view.findViewById(R.id.setPersonalInfo_button_userProfileCreation)
+        val inputFirstnameText: EditText =
+            view.findViewById(R.id.input_firstname_text_UserProfileCreation)
+        val inputSurnameText: EditText =
+            view.findViewById(R.id.input_surname_text_UserProfileCreation)
+        validatePersonalInfoButton.setOnClickListener {
+            //check the firstname
+            firstname = inputFirstnameText.text.toString()
+            val test1 = checkName(firstname, errorFirstnameText)
+
+            //check the surname
+            surname = inputSurnameText.text.toString()
+            val test2 = checkName(surname, errorSurnameText)
+
+            val test3 = checkDateOfBirth(dateOfBirth, errorDateText)
+
+            //if the data are correct, set them to the database and show the next fragment when click on the validate button
+            if (test1 && test2 && test3) {
+                database.setPersonalInfo(userName, firstname, surname, dateOfBirth)
+                val previousActivity = activity
+                if (previousActivity != null) {
+                    val fragManagement = previousActivity.supportFragmentManager.beginTransaction()
+                    val dataToUserGoalsInitFrag: Bundle = Bundle()
+                    //data to transmit to the UserGoalsInitFragment(username + isTest)
+                    dataToUserGoalsInitFrag.putBoolean("isRunningTestForDataBase", isTest)
+                    dataToUserGoalsInitFrag.putString("userName", userName)
+                    val userGoalsInitFrag = UserGoalsInitFragment()
+                    userGoalsInitFrag.arguments = dataToUserGoalsInitFrag
+                    fragManagement.replace(R.id.personalInfoFragment, userGoalsInitFrag).commit()
+                }
+            }
+        }
+        return view
+    }
+}
+
+/**
+ * Helper function to check if a name is in the correct format
+ * @param name to be checked
+ * @param outputErrorText text on the UI used to print the existence of error to the user
+ * @return true if the name is in the correct format, false otherwise
+ */
+private fun checkName(name: String, outputErrorText: TextView): Boolean {
+    //error messages that can be displayed in case of wrong input on click on validate
+    val emptyName: String = "* This field can't be empty !"
+    val incorrectName: String =
+        "* This field is in an incorrect format ! It must be composed of letters or character '-'"
+    //check if the name is empty
+    if (name == "") {
+        outputErrorText.text = emptyName
+        outputErrorText.setTextColor(Color.RED)
+        return false
+    }
+    //check all the characters of the name
+    val tabChar: CharArray = name.toCharArray()
+    for (i in 0..(name.length - 1)) {
```

Comment:
The following one-liner could replace the need of the for loop and the char array conversion:

```kotlin
if (name.find { !it.isLetter() && it != '-' }) {
   // Handle error
}
```

Reply:
Thank you very much for this line, I didn't know that.